aspnetcore/blazor/components/prerender.md

```diff
@@ -103,7 +103,28 @@ No compile time error occurs, but a runtime error occurs during prerendering:
 
 This error occurs because the component must compile and execute on the server during prerendering, but <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment> isn't a registered service on the server.
 
-If the app doesn't require the value during prerendering, this problem can be solved by injecting <xref:System.IServiceProvider> to obtain the service instead of the service type itself:
+There are several approaches that you can take to address this scenario:
+
+* If the service supports server execution, register the service on the server in addition to the client so that it's available during prerendering. For an example of this scenario, see the guidance for <xref:System.Net.Http.HttpClient> services in the [Blazor Web App external web APIs](xref:blazor/call-web-api#blazor-web-app-external-web-apis) section of the *Call web API* article.
+
+* Make the service optional if it isn't always needed during prerendering.
+
+* If a different service implementation is needed on the server, create a service abstraction and create implementations for the service in the `.Client` and server projects. Register the services in each project. Inject the custom service abstraction in the component.
+
+* Disable prerendering for the component or for the entire app.
+
+To optionally inject the service into the component, use constructor injection:
```

Comment:
I think showing the constructor injection approach is simpler than using `IServiceProvider`, so I think it should be suggested first.